nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -457,198 +153,322 @@ def forward(self, obj):
-    how to get the scalar type from torch._C.TensorType.
+    if ivalue is None or isinstance(ivalue, torch.memory_format):
+        return ivalue
+    elif isinstance(ivalue, int):
+        global enum_to_memory_format_dict
+        if ivalue in enum_to_memory_format_dict:
+            return enum_to_memory_format_dict[ivalue]
+    raise TypeError('No torch.memory_format corresponding to the value \'%s\'', ivalue)
+
+special_treat_dict = {
+    'dtype': dtype_trans,
+    'memory_format': memory_format_trans,
+}
+schema_fix_dict = {
+    # functinon 'to', 'randint', and 'sparse_coo_tensor' has different schema between python and c++.
+    # https://pytorch.org/docs/stable/jit_unsupported.html#ops-with-divergent-schemas-between-torch-python
+    'aten::to.device(Tensor(a) self, Device device, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))':
+        'aten::to.device(Tensor(a) self, Device device, int dtype, bool non_blocking=False, bool copy=False, *, int? memory_format=None) -> (Tensor(a))',
+    'aten::to.dtype(Tensor(a) self, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))':
+        'aten::to.dtype(Tensor(a) self, int dtype, bool non_blocking=False, bool copy=False, *, int? memory_format=None) -> (Tensor(a))',
+    'aten::to.other(Tensor(a) self, Tensor other, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))':
+        'aten::to.other(Tensor(a) self, Tensor other, bool non_blocking=False, bool copy=False, *, int? memory_format=None) -> (Tensor(a))',
+    
+    # todo: are the arguments 'pin_memory' and 'requires_grad' related? functions in the python have only 'requires_grad' and functions in the aten have only 'pin_memory'
+    
+    # 'aten::randint(int high, int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)',
+    # 'aten::randint.generator(int high, int[] size, *, Generator? generator, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)',
+    # 'aten::randint.low(int low, int high, int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)',
+    # 'aten::randint.low_generator(int low, int high, int[] size, *, Generator? generator, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)',
+
+    # 'aten::sparse_coo_tensor.size(int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=False) -> (Tensor)',
+    # 'aten::sparse_coo_tensor.indices(Tensor indices, Tensor values, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)',
+    # 'aten::sparse_coo_tensor.indices_size(Tensor indices, Tensor values, int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)'
+}
+@lru_cache(maxsize=256)
```

Comment:
I tried the official `torch._C.parse_schema` and it's a little slower than writing the parser manually. It will cost about 3 second to parse 10000 schema. If the schema repeats, using an `lru_cache` only cost less than 0.1 seconds, and some models do have many same schemas.